genesis/engine/solvers/rigid/rigid_solver_decomp.py

```diff
@@ -2296,6 +2296,49 @@ def update_verts_for_geom(self, i_g):
             self.fixed_verts_state,
         )
 
+    def get_weld_constraints(self, as_tensor: bool = True, to_torch: bool = True):
+        n_eqs = tuple(self.constraint_solver.constraint_state.ti_n_equalities.to_numpy())
+        n_envs = len(n_eqs)
+        n_max = max(n_eqs) if n_eqs else 0
+
+        if as_tensor:
+            out_size = n_envs * n_max
+            splits = None
+        else:
+            cumsum = np.cumsum(n_eqs, dtype=np.int32)
+            splits = list(cumsum[:-1])
+            out_size = int(cumsum[-1]) if n_envs else 0
+
+        if to_torch:
+            buf = torch.full((out_size, 3), -1, dtype=gs.tc_int, device=gs.device)
+        else:
+            buf = np.full((out_size, 3), -1, dtype=np.int32)
+
+        if n_max > 0:
+            kernel_collect_welds(
+                as_tensor,
+                buf,
+                self.constraint_solver.constraint_state,
+                self.equalities_info,
+                self._static_rigid_sim_config,
+            )
+
+        if as_tensor:
+            if n_envs > 0:
+                buf = buf.view(n_envs, n_max, 3) if to_torch else buf.reshape(n_envs, n_max, 3)
+            env_idx, obj_a, obj_b = buf[..., 0], buf[..., 1], buf[..., 2]
+            return {"env": env_idx, "obj_a": obj_a, "obj_b": obj_b}
```

Comment:
Adding key `"env"` should not be necessary. It was not there in `Collider.get_contacts`.